base/applications/winver/osinfo.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * PROJECT:    ReactOS Version Program
+ * LICENSE:    LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:    Retrieve OS name and simple compatibility information
+ * COPYRIGHT:  Copyright 2025 Thamatip Chitpong <thamatip.chitpong@reactos.org>
+ */
+
+#include "winver_p.h"
+
+#define OSINFO_KEY L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion"
+#define MAJOR_VER_MAX_LEN 16
+#define MINOR_VER_MAX_LEN 16
+#define BUILD_NUMBER_MAX_LEN 16
+
+static
+LPWSTR
+Winver_GetRegValueString(
+    _In_ HKEY hKey,
+    _In_ LPCWSTR lpValue)
+{
+    DWORD dwType = REG_NONE;
```

Comment:
If `RegQueryValueExW` successful, `dwType` will be set.

```suggestion
    DWORD dwType;
```

Reply:
OK.